docs/tutorials/12_examples_knapsack_problem.ipynb

```diff
@@ -0,0 +1,817 @@
+        "\n",
+        "### Investment problem\n",
+        "Considering the above characteristic of the mountaineer in the context of business instead of leisure leads to a second classical interpretation of KP as an investment problem. A wealthy individual or institutional investor has a certain amount\n",
+        "of money $c$ available which she wants to put into profitable business projects. As a basis for her decisions, she compiles a long list of possible investments, the required amount $w_j$, and the expected net return $p_j$ over a fixed period. The aspect of risk is not explicitly taken into account here. Obviously, the combination of the binary decisions for every investment such that the overall return on investment is as large as possible can be formulated by KP.\n",
+        "\n",
+        "### Airline cargo business\n",
+        "The dispatcher of a cargo airline has to decide which of the transportation requests posed by the customers he should fulfill, i.e. how to load a particular plane. His decision is based on a list of requests which contain the weight $w_j$ of every package and the rate per weight unit charged for each request. Note that this rate is not fixed, but depends on the particular long-term arrangements with every customer. Hence, the profit $p_j$ made by the company by accepting a request and by putting the corresponding package on the plane is not directly proportional to the weight of the package. Naturally, every plane has a specified maximum capacity $c$ which may not be exceeded by the total weight of the selected packages. This logistic problem is a direct analog to the packing of the mountaineers' knapsack.\n",
+        "\n",
+        "### Log cutting\n",
+        "While the previous examples all contain elements of \"packing\", one may also view the KP as a \"cutting\" problem. Assume that a sawmill has to cut a log into shorter pieces. The pieces must however be cut into some predefined standard-lengths $w_j$, where each length has an associated selling price $p_j$. In order to maximize the profit of the log, the sawmill can formulate the problem as a KP where the length of the log defines the capacity $c$.\n",
+        "\n",
+        "## Variants of the knapsack problem\n",
+        "\n",
+        "Let us consider again the previous problem of the cargo airline dispatcher. In a different setting, the profit made by accepting a package may be directly proportional to its weight. In this case, the optimal loading of the plane is achieved by filling as much weight as possible into the plane, or equivalently setting $p_j = w_j$ in KP. The resulting optimization problem is known as the **Subset Sum Problem (SSP)** because we are looking for a subset of the values $w_j$ with the sum being as close as possible to, but not exceeding, the given target value $c$. Replacing constrain (1) in KP we can define SSP: $\\texttt{maximize} \\; \\sum_{j=1}^{N}w_jx_j$.\n",
```

Comment:
`Replacing constrain (1) in KP we can define SSP` should be `Replacing the objective function (1) in KP we can define SSP`. (1) is not a constraint but the objective function.

Reply:
That's right! Thanks for noticing, I've updated it!